File: Domino/Domino/BlockDomino.cpp
#include "BlockDomino.h"

#include <ctime>


BlockDomino::BlockDomino(int value_up=-1, int value_down=-1)
{
	if(value_up>=0 && value_up<=6)
		this->value_up = value_up;
	else
		this->value_up = 0;

	if(value_down>=0 && value_down<=6)
		this->value_down = value_down;
	else
		this->value_down = 0;

	rotate = no_rotate;

}
BlockDomino::BlockDomino()
{
	value_up   = 0;
	value_down = 0;

	rotate = no_rotate;
}

BlockDomino::~BlockDomino(void)
{
}

void BlockDomino::set_value_up(int value_up)
{
	if(value_up>=0 && value_up<=6)
		this->value_up = value_up;
	else
		this->value_up = 0;
}
int BlockDomino::get_value_up()
{
	return value_up;
}

void BlockDomino::set_value_down(int value_down)
{
	if(value_down>=0 && value_down<=6)
		this->value_down = value_down;
	else
		this->value_down = 0;
}
int BlockDomino::get_value_down()
{
	return value_down;
}

void BlockDomino::set_rotate(::rotate rotate)
{
	this->rotate = rotate;
}
::rotate BlockDomino::get_rotate()
{
	return rotate;
}

File: Domino C#/Source/Domino/BlockDomino.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace DominoGame
{
    class BlockDomino
    {
        private int value_up;   //The value of up domino
        private int value_down; //the value of down domino
        private rotate rotate; //the position of domino
        
        public BlockDomino() //default -> not use!
        {
            value_down = 0;  
            value_up = 0;
        }

        public BlockDomino(int value_up, int value_down) //make a block
        {
            if (value_down >= 0 && value_down <= 6)   //value have to be in <0;6> -> like DOMINO :D
                this.value_down = value_down;
            else
                this.value_down = 0;

            if (value_up >= 0 && value_up <= 6)
                this.value_up = value_up;
            else
                this.value_up = 0;

            rotate = rotate.no_rotate;   //at start rotate is needless
        }
        
        //Setters And Getters
        public int Value_Up
        {
            get
            {
                return value_up;
            }
            set
            {
                value_up = value;
            }
        }

        public int Value_Down
        {
            get
            {
                return value_down;
            }
            set
            {
                value_down = value;
            }
        }

        public rotate Rotate
        {
            get
            {
                return rotate;
            }
            set
            {
                rotate = value;
            }
        }

    }
}
